tasks/test-project/tui-tasks.js

```diff
@@ -406,38 +408,76 @@ async function apiTasks(outputPath, { linkWithLatestFwBuild }) {
-    // to work. So we update the package.json to make the setup use the latest
-    // canary version for the api and web sides
+    // We want to use the latest version of the auth-dbauth-{setup,api,web}
+    // packages. But they're not published yet. So let's package them up as
+    // tarballs and install them using that by setting yarn resolutions
 
-    const { stdout } = await exec(
-      `yarn npm info @redwoodjs/auth-dbauth-setup --fields versions --json`,
-      [],
-      execaOptions
+    const setupPkg = path.join(
+      RW_FRAMEWORK_PATH,
+      'packages',
+      'auth-providers',
+      'dbAuth',
+      'setup'
+    )
+    const apiPkg = path.join(
+      RW_FRAMEWORK_PATH,
+      'packages',
+      'auth-providers',
+      'dbAuth',
+      'api'
     )
+    const webPkg = path.join(
+      RW_FRAMEWORK_PATH,
+      'packages',
+      'auth-providers',
+      'dbAuth',
+      'web'
+    )
+
+    await exec('yarn build:pack', [], getExecaOptions(setupPkg))
+    await exec('yarn build:pack', [], getExecaOptions(apiPkg))
+    await exec('yarn build:pack', [], getExecaOptions(webPkg))
 
-    const latestCanaryVersion = JSON.parse(stdout)
-      .versions.filter((version) => version.includes('canary'))
-      .at(-1)
+    const setupTgz = path.join(setupPkg, 'redwoodjs-auth-dbauth-setup.tgz')
+    const apiTgz = path.join(apiPkg, 'redwoodjs-auth-dbauth-api.tgz')
+    const webTgz = path.join(webPkg, 'redwoodjs-auth-dbauth-web.tgz')
 
-    const dbAuthSetupPath = path.join(
+    const setupTgzDest = path.join(
+      outputPath,
+      'node_modules',
+      'auth-dbauth-setup.tgz'
+    )
+    const apiTgzDest = path.join(
       outputPath,
       'node_modules',
-      '@redwoodjs',
-      'auth-dbauth-setup'
+      'auth-dbauth-api.tgz'
     )
+    const webTgzDest = path.join(
+      outputPath,
+      'node_modules',
+      'auth-dbauth-web.tgz'
+    )
+
+    fs.copyFileSync(setupTgz, setupTgzDest)
+    fs.copyFileSync(apiTgz, apiTgzDest)
+    fs.copyFileSync(webTgz, webTgzDest)
 
-    const dbAuthSetupPackageJson = JSON.parse(
-      fs.readFileSync(path.join(dbAuthSetupPath, 'package.json'), 'utf-8')
+    const projectPackageJsonPath = path.join(outputPath, 'package.json')
+    const projectPackageJson = JSON.parse(
+      fs.readFileSync(projectPackageJsonPath, 'utf-8')
     )
 
-    dbAuthSetupPackageJson.version = latestCanaryVersion
+    projectPackageJson.resolutions ??= {}
+    projectPackageJson.resolutions = {
+      ...projectPackageJson.resolutions,
+      'auth-dbauth-setup': './node_modules/auth-dbauth-setup.tgz',
+      'auth-dbauth-api': './node_modules/auth-dbauth-api.tgz',
+      'auth-dbauth-web': './node_modules/auth-dbauth-web.tgz',
+    }
 
     fs.writeFileSync(
-      path.join(dbAuthSetupPath, 'package.json'),
-      JSON.stringify(dbAuthSetupPackageJson, null, 2)
+      projectPackageJsonPath,
+      JSON.stringify(projectPackageJson, null, 2)
```

Comment:
Not a comment on this line in particular, but after this line we run `exec('yarn rw setup auth dbAuth')`. Unless we run `yarn install` first, the tarballs won't be unpacked into `node_modules` and `yarn rw setup` probably won't use the new code

Reply:
Good catch. `yarn rw setup` does `yarn install` for the web and api packages. So the new version of those would have been used I think. But not the new version of the setup package itself.